common/src/main/java/com/box/l10n/mojito/notifications/service/ThirdPartyNotificationService.java

```diff
@@ -0,0 +1,16 @@
+package com.box.l10n.mojito.notifications.service;
+
+import com.google.common.collect.ImmutableMap;
+import com.ibm.icu.text.MessageFormat;
+
+public abstract class ThirdPartyNotificationService {
+    
+    public abstract void sendMessage(String messageTemplate, ImmutableMap<String, String> serviceParameters) throws Exception;
```

Comment:
After looking at it again I agree, I'll clean this up so that a NotificationService.sendNotifications accepts just a message string & a set of services to notify with that message. If the messages are to differ across different service types then the calling object will need to handle that by calling the notification service with different parameters